src/util/channel_table.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * channel_table.c
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <common_table.h>
+#include <channel_table.h>
+#include <charbuf.h>
+#include <pelz_enclave_log.h>
+
+#include "sgx_trts.h"
+#include ENCLAVE_HEADER_TRUSTED
+#include "sgx_retrieve_key_impl.h"
+
+ChanTable chan_table = {
+  .chan_key = NULL,
+  .num_entries = 0
+};
+
+TableResponseStatus chan_table_init(int entry_num)
+{
+  chan_table.num_entries = entry_num;
+  
+  charbuf *temp;
+
+  if ((temp = (charbuf *) malloc(entry_num * sizeof(charbuf))) == NULL)
+  {
+    pelz_log(LOG_ERR, "Channel Key List Space Allocation Error");
+    return ERR_REALLOC;
```

Comment:
```suggestion
    return MEM_ALLOC_FAIL;
```

Seems more accurate here.